components/automate-ui/src/app/pages/project/rules/project-rules.component.html

```diff
@@ -0,0 +1,110 @@
+          <label>
+            <span class="label">Rule Name</span>
+            <chef-input id="ruleName" ngDefaultControl formControlName="name"></chef-input>
+          </label>
+          <chef-error
+            *ngIf="ruleForm.get('name').hasError('required') && ruleForm.get('name').dirty"
+          >Name is required</chef-error>
+        </chef-form-field>
+
+        <chef-form-field class="display3">
+          <label>
+            <span class="label">Resource Type</span>
+            <select id="resourceType" ngDefaultControl formControlName="type">
+              <option value='NODE'>Node</option>
+              <option value='EVENT'>Event</option>
+            </select>
+          </label>
+          <chef-error
+            *ngIf="ruleForm.get('type').hasError('required') && ruleForm.get('type').dirty"
+          >Resource Type is required</chef-error>
+          <small class="help">Resource types cannot be changed after the rule is created.</small>
+        </chef-form-field>
+
+        <div id="add-condition">
+          <div class="label">Conditions</div>
+          <small>All conditions must be true for the rule to be true.</small>
+          <chef-button primary (click)="addCondition()">Add Condition</chef-button>
+        </div>
+
+        <div id="condition-list" formArrayName="conditions">
+
+          <div class="condition-item"
+            [formGroupName]="i"
+            *ngFor="let condition of ruleForm.get('conditions').controls; let i = index">
+
+            <chef-form-field class="attribute-field">
+              <label>
+                <span class="label">{{ getAttributeLabel() }}</span>
+                <select ngDefaultControl formControlName="attribute">
+                  <option 
+                    *ngFor="let option of attributes[ruleForm.get('type').value.toLowerCase()]; let i = index"
+                    [value]="option.key"
+                  >{{ option.value }}</option>
+                </select>
+              </label>
+            </chef-form-field>
+
+            <chef-form-field class="attribute-field">
+              <label>
+                <span class="label">Operator</span>
+                <select ngDefaultControl
+                  formControlName="operator"
+                  (change)="updateConditionValues(condition)"
+                >
+                  <option value='EQUALS'>equals</option>
+                  <option value='MEMBER_OF'>member of</option>
+                </select>
+              </label>
+            </chef-form-field>
+            
+            <chef-form-field class="attribute-field">
+              <label>
+                <span class="label">Value</span>
+                <chef-input ngDefaultControl
+                  formControlName="values"
+                  [value]="getConditionValue(condition.controls.values.value)"
+                  (change)="updateConditionValues(condition)"
+                ></chef-input>
+              </label>
```

Comment:
Need `<chef-error>` element here with a check for required and non-blank pattern:
```
<chef-error *ngIf="(ruleForm.get('values').hasError('required') || ruleForm.get('values').hasError('pattern')) && ruleForm.get('values').dirty">
Value is required.
</chef-error>
```
The problem, however, is that this field is in a list, so the name `values` won't work (and won't work in L79 above either). I am not sure how to do the validation for a list element like this.